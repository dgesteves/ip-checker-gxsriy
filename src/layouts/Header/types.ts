import { IWookieApiResponse } from "../../services/wookieApi/types";

export default interface IHeaderProps {
  IPData: IWookieApiResponse | undefined;
  onSearch: (IP: string) => void;
}
