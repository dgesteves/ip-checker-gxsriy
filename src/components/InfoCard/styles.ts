import styled from "styled-components";

export const StyledInfoCard = styled.section`
  position: absolute;
  bottom: -15%;
  z-index: 1000;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: ${({ theme }) => theme.colors.white};
  width: ${({ theme }) => `calc(${theme.sizes.s100}*10)`};
  height: ${({ theme }) => theme.sizes.s100};
  border-radius: ${({ theme }) => theme.sizes.s8};
  padding: ${({ theme }) => theme.sizes.s24};
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    flex-direction: column;
    width: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
    height: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
    bottom: -55%;
  }
`;

export const StyledInfoCardElement = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${({ theme }) => theme.sizes.s8};

  :not(:first-child) {
    border-left: 1px solid ${({ theme }) => theme.colors.gray};
    padding-left: ${({ theme }) => theme.sizes.s18};
  }

  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    align-items: center;
    :not(:first-child) {
      border-left: none;
      padding-left: 0;
    }
  }
`;

export const StyledInfoCardElementTitle = styled.h4`
  font-size: ${({ theme }) => theme.sizes.s12};
  font-weight: 500;
  text-transform: uppercase;
`;

export const StyledInfoCardElementValue = styled.h3`
  color: ${({ theme }) => theme.colors.very_dark_gray};
  font-size: ${({ theme }) => theme.sizes.s18};
  text-align: center;
`;
