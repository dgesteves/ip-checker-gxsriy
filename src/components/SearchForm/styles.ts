import styled from "styled-components";

export const StyledForm = styled.form`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${({ theme }) => `calc(${theme.sizes.s100}*5)`};
  height: ${({ theme }) => theme.sizes.s44};
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    width: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
    touch-action: manipulation;
  }
`;

export const StyledInput = styled.input`
  width: 100%;
  height: 100%;
  border: none;
  border-top-left-radius: ${({ theme }) => theme.sizes.s8};
  border-bottom-left-radius: ${({ theme }) => theme.sizes.s8};
  padding: ${({ theme }) => theme.sizes.s16};
  font-size: ${({ theme }) => theme.sizes.s18};
  outline: none;
`;

export const StyledButton = styled.button`
  height: 100%;
  width: ${({ theme }) => theme.sizes.s44};
  border: none;
  outline: none;
  border-top-right-radius: ${({ theme }) => theme.sizes.s8};
  border-bottom-right-radius: ${({ theme }) => theme.sizes.s8};
  background-color: ${({ theme }) => theme.colors.very_dark_gray};
  cursor: pointer;
  transition: all 0.4s ease-in-out;

  :hover {
    width: ${({ theme }) => theme.sizes.s64};
  }
`;

export const StyledErrorMessage = styled.p`
  color: ${({ theme }) => theme.colors.error};
  font-size: ${({ theme }) => theme.sizes.s18};
  font-weight: 500;
  text-shadow: 0 0 80px rgb(192 219 255 / 35%), 0 0 32px rgb(65 120 255 / 24%);
`;
