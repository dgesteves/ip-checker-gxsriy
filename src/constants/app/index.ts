export const HEADER_TITLE = "IP Address Tracker";
export const SEARCH_PLACEHOLDER = "Search for any IP address or domain";
export const TIME_STANDARD = "UTC";
export const IPIFY_API_URL = "https://api.ipify.org";
export const WOOKIE_BASE_API_URL = "https://wookie.codesubmit.io/ipcheck";
export const LOCALHOST_IP = "127.0.0.1";
export const MAP_FALLBACK_TEXT = "No location found for this IP address";
export const MAP_FALLBACK_LAT = 0;
export const MAP_FALLBACK_LNG = 0;
export const MAP_ZOOM = 17;
export const VALID_IP_ADDRESS =
  /^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;

export enum INFO_CARD_ELEMENTS {
  IP = "ip address",
  LOCATION = "location",
  TIMEZONE = "timezone",
  ISP = "isp",
}

export enum MAP_TILE_LAYER {
  ATTRIBUTION = '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors',
  URL = "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
}
