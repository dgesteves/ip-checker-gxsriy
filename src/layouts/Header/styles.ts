import styled from "styled-components";

export const StyledHeader = styled.header<{ img: string }>`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: ${({ theme }) => theme.sizes.s24};
  height: 30vh;
  padding: ${({ theme }) => theme.sizes.s24};
  background-image: url(${({ img }) => img});
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
`;

export const StyledHeaderTitle = styled.h1`
  color: ${({ theme }) => theme.colors.white};
  font-size: ${({ theme }) => theme.sizes.s28};
  font-weight: 500;
}
  `;
