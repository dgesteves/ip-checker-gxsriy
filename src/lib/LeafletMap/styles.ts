import styled from "styled-components";
import { MapContainer } from "react-leaflet";

export const StyledMapContainer = styled(MapContainer)`
  height: 70vh;
`;
