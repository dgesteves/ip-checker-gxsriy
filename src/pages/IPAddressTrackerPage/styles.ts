import styled from "styled-components";

export const StyledMapFallback = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 70vh;
`;

export const StyledMapFallbackText = styled.p`
  background-color: ${({ theme }) => theme.colors.error};
  color: ${({ theme }) => theme.colors.white};
  padding: ${({ theme }) => theme.sizes.s8} ${({ theme }) => theme.sizes.s16};
  border-radius: ${({ theme }) => theme.sizes.s8};
  font-size: ${({ theme }) => theme.sizes.s24};
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    width: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
  }
`;
