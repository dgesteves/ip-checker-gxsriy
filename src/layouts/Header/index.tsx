import { StyledHeader, StyledHeaderTitle } from "./styles";
import background from "../../assets/img/optimized-pattern-bg.png";
import SearchForm from "../../components/SearchForm";
import InfoCard from "../../components/InfoCard";
import IHeaderProps from "./types";
import { HEADER_TITLE } from "../../constants/app";

export default function Header({ IPData, onSearch }: IHeaderProps) {
  return (
    <StyledHeader img={background}>
      <StyledHeaderTitle>{HEADER_TITLE}</StyledHeaderTitle>
      <SearchForm onSearch={onSearch} searchError={IPData?.messages} />
      {!IPData?.messages && <InfoCard IPData={IPData} />}
    </StyledHeader>
  );
}
