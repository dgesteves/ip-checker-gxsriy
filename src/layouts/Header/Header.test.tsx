import { render, screen } from "@testing-library/react";
import { theme } from "../../assets/styles/theme";
import { ThemeProvider } from "styled-components";
import GlobalStyles from "../../assets/styles/globalStyles";
import Header from "./index";
import { GETIPDATA_RESPONSE } from "../../constants/tests";
import { HEADER_TITLE } from "../../constants/app";

describe("Header component", () => {
  const onSearch = jest.fn();

  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <Header IPData={GETIPDATA_RESPONSE} onSearch={onSearch} />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders correctly with no data", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <Header IPData={undefined} onSearch={onSearch} />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <Header IPData={GETIPDATA_RESPONSE} onSearch={onSearch} />
      </ThemeProvider>
    );

    expect(container).toBeInTheDocument();
  });

  it("should render the header title", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <Header IPData={GETIPDATA_RESPONSE} onSearch={onSearch} />
      </ThemeProvider>
    );

    expect(screen.getByText(HEADER_TITLE)).toBeInTheDocument();
  });

  it("should render the SearchForm component", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <Header IPData={GETIPDATA_RESPONSE} onSearch={onSearch} />
      </ThemeProvider>
    );

    expect(screen.getByTestId("search-form")).toBeInTheDocument();
  });

  it("should render the InfoCard component", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <Header IPData={GETIPDATA_RESPONSE} onSearch={onSearch} />
      </ThemeProvider>
    );

    expect(screen.getByTestId("info-card")).toBeInTheDocument();
  });
});
