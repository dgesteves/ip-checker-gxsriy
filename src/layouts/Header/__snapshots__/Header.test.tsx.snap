// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div>
  <header
    class="sc-bczRLJ eSXdNZ"
  >
    <h1
      class="sc-gsnTZi ijdOdb"
    >
      IP Address Tracker
    </h1>
    <form
      class="sc-dkzDqf ebTfFI"
      data-testid="search-form"
    >
      <input
        class="sc-hKMtZM iDhKft"
        placeholder="Search for any IP address or domain"
        type="text"
        value=""
      />
      <button
        class="sc-eCYdqJ bicgSB"
        type="submit"
      >
        <svg
          height="14"
          width="11"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m2 1 6 6-6 6"
            fill="none"
            stroke="#fff"
            stroke-width="3"
          />
        </svg>
      </button>
    </form>
    <section
      class="sc-gKXOVf urLmO"
      data-testid="info-card"
    >
      <div
        class="sc-iBkjds kPtZCg"
      >
        <h4
          class="sc-ftvSup gLiHex"
        >
          ip address
        </h4>
        <h3
          class="sc-papXJ fuZpkM"
        >
          111.111.111.111
        </h3>
      </div>
      <div
        class="sc-iBkjds kPtZCg"
      >
        <h4
          class="sc-ftvSup gLiHex"
        >
          location
        </h4>
        <h3
          class="sc-papXJ fuZpkM"
        >
          Uchikanda, JP 
        </h3>
      </div>
      <div
        class="sc-iBkjds kPtZCg"
      >
        <h4
          class="sc-ftvSup gLiHex"
        >
          timezone
        </h4>
        <h3
          class="sc-papXJ fuZpkM"
        >
          UTC
           
          +09:00
        </h3>
      </div>
      <div
        class="sc-iBkjds kPtZCg"
      >
        <h4
          class="sc-ftvSup gLiHex"
        >
          isp
        </h4>
        <h3
          class="sc-papXJ fuZpkM"
        >
          KDDI CORPORATION
        </h3>
      </div>
    </section>
  </header>
</div>
`;

exports[`Header component renders correctly with no data 1`] = `
<div>
  <header
    class="sc-bczRLJ eSXdNZ"
  >
    <h1
      class="sc-gsnTZi ijdOdb"
    >
      IP Address Tracker
    </h1>
    <form
      class="sc-dkzDqf ebTfFI"
      data-testid="search-form"
    >
      <input
        class="sc-hKMtZM iDhKft"
        placeholder="Search for any IP address or domain"
        type="text"
        value=""
      />
      <button
        class="sc-eCYdqJ bicgSB"
        type="submit"
      >
        <svg
          height="14"
          width="11"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m2 1 6 6-6 6"
            fill="none"
            stroke="#fff"
            stroke-width="3"
          />
        </svg>
      </button>
    </form>
  </header>
</div>
`;
